test/e2e/fixture-builder.js

```diff
@@ -229,15 +229,9 @@ class FixtureBuilder {
   }
 
   withNetworkControllerOnMainnet() {
-    merge(this.fixture.data.NetworkController, {
-      providerConfig: {
-        chainId: CHAIN_IDS.MAINNET,
-        nickname: '',
-        rpcUrl: '',
-        type: NETWORK_TYPES.MAINNET,
-      },
+    return this.withNetworkController({
+      providerConfig: { id: 'mainnet' },
```

Comment:
The occurrences of `providerConfig` must remain in this file for e2e tests to pass.

Why?  It seems like this e2e state runs through all state migrations.   If I use the new pattern with  `selectedNetworkClientId`, migration 92.2 removes it.  It used to get put back by migration 94, but that pulls it from `providerConfig`.  So keeping `providerConfig` lets us get to the right state in the end.

It feels like maybe e2e should just use the state we provide without running all migrations on it.

Reply:
Good eye, I hadn't noticed `FIXTURE_STATE_METADATA_VERSION` is the reason for this